### PR TITLE
ceph-nfs: run rados cmds in container

### PR DESCRIPTION
**Pull request by tombarron**
When ceph-nfs is deployed containerized and ceph-common is not
installed on the host the start_nfs task fails because the rados
command is missing on the host.

Run rados commands from a ceph container instead so that
they will succeed.

Signed-off-by: Tom Barron <tpb@dyncloud.net>

### CODE REVIEW

File: roles/ceph-nfs/tasks/start_nfs.yml [hunk 1]
```diff
@@ -1,6 +1,12 @@
 ---
+- name: set_fact docker_exec_cmd_nfs
+  set_fact:
+    docker_exec_cmd_nfs: "docker exec ceph-mon-{{ hostvars[groups[mon_group_name][0]]['ansible_hostname'] }}"
+  when:
+    - containerized_deployment
+
 - name: check if rados index object exists
-  shell: "rados -p {{ cephfs_data }} --cluster {{ cluster }} ls|grep {{ ceph_nfs_rados_export_index }}"
+  shell: "{{ docker_exec_cmd_nfs | default('') }} rados -p {{ cephfs_data }} --cluster {{ cluster }} ls|grep {{ ceph_nfs_rados_export_index }}"
```
Comment (leseb) [inline roles/ceph-nfs/tasks/start_nfs.yml]: If you want to execute this command on a monitor you also need to `delegate_to: "{{ groups[mon_group_name][0] }}"` on the task

File: roles/ceph-nfs/tasks/start_nfs.yml [hunk 2]
```diff
@@ -9,7 +15,7 @@
     - ceph_nfs_rados_backend
 
 - name: create an empty rados index object
-  shell: "echo | rados -p {{ cephfs_data }} --cluster {{ cluster }} put {{ ceph_nfs_rados_export_index }} -"
+  shell: "{{ docker_exec_cmd_nfs | default('') }} rados -p {{ cephfs_data }} --cluster {{ cluster }} put {{ ceph_nfs_rados_export_index }} /dev/null"
```
Comment (leseb) [inline roles/ceph-nfs/tasks/start_nfs.yml]: ditto

File: roles/ceph-nfs/tasks/start_nfs.yml (hunk 1 above)
Comment (tombarron) [reply roles/ceph-nfs/tasks/start_nfs.yml]: ok

File: roles/ceph-nfs/tasks/start_nfs.yml (hunk 2 above)
Comment (tombarron) [reply roles/ceph-nfs/tasks/start_nfs.yml]: the delegation is already present for this task.

**Pull request merged**